### PR TITLE
Update contributing guide

### PR DESCRIPTION
**Pull request by michael-valdron**
## What does this PR do?

Updates contributing guide and supporting documents with up to date information and layout.

Additionally, adds missing DCO file into repository.

### Which issue(s) does this PR fix

fixes https://github.com/devfile/api/issues/1346

### PR acceptance criteria

Testing and documentation do not need to be complete in order for this PR to be approved. We just need to ensure tracking issues are opened.

> - Open new test/doc issues under the [devfile/api](https://github.com/devfile/api/issues) repo
> - Check each criteria if:
> - There is a separate tracking issue. Add the issue link under the criteria
>  **or**
> - test/doc updates are made as part of this PR
> - If unchecked, explain why it's not needed

- [ ] Unit/Functional tests

  

- [ ] [QE Integration test](https://github.com/devfile/integration-tests)

  

- [X] Documentation

   

- [ ] Client Impact

### CODE REVIEW

**Comment by michael-valdron** [comment]
> Client Impact

N/A

File: CONTRIBUTING.md
Comment (yangcao77) [inline CONTRIBUTING.md:88]: https://kubernetes.slack.com/archives/C02SX9E5B55

File: CONTRIBUTING.md
Comment (thepetk) [inline CONTRIBUTING.md:41]: Might be good to mention that for the `devfile/api` reviews from `CODEOWNERS` are required?

**Review by thepetk — COMMENTED** [review]
Left a minor comment

File: CONTRIBUTING.md
Comment (michael-valdron) [reply CONTRIBUTING.md:14]: fixed in https://github.com/devfile/api/pull/1597/commits/fd6dd3fbdea06de888af1d8d6f0b3c7c03ec18ab, missed that this repository was also missing the `DCO` file.

File: CONTRIBUTING.md
Comment (michael-valdron) [reply CONTRIBUTING.md:41]: @thepetk @yangcao77 How is this change? https://github.com/devfile/api/pull/1597/commits/bd2157e65d7a156e489c940927d463b0efaba072

File: CONTRIBUTING.md
Comment (michael-valdron) [reply CONTRIBUTING.md]: @yangcao77 @thepetk How is this change? https://github.com/devfile/api/pull/1597/commits/aa7e5b38212f2c766d9e9cbb1758cfde3a37c4c6

**Comment by michael-valdron** [comment]
> https://github.com/devfile/api/pull/1597#discussion_r1640042718
> https://github.com/devfile/api/pull/1597#discussion_r1640117835

The follow up changes for the feedback above should be include in the contributing guide template #1591 and updated in all repositories to be consistent.

FYI @yangcao77 @thepetk

**Review by thepetk — APPROVED** [review]
/lgtm after @yangcao77 discussions are resolved

**Review by yangcao77 — APPROVED** [review]
lgtm

**Pull request merged**
